gs_bucket_backup/bucket_sync.py

```diff
@@ -0,0 +1,168 @@
+        self.suffix = file_suffix
+
+    def _subprocess_debug_wrap(self,
+                               cmd: list,
+                               shellmode: bool = False) -> str:
+
+        # Sanitize out potential output of encryption key
+        debug_cmd_str = "Calling command {}".format(
+            " ".join(cmd)).replace(self.encrypt_key, "X"*10)
+        logger.debug(debug_cmd_str)
+
+        if shellmode:
+            cmd = " ".join(cmd)
+        copy_output = subprocess.check_output(
+            cmd, stderr=subprocess.STDOUT, shell=shellmode).decode('ascii')
+        logger.debug(copy_output)
+
+        return copy_output
+
+    def initialize_svc_acct(self, acct_key_path: str) -> None:
+        """ Initilize gcloud tooling using a GCP service account key """
+        gcloud_auth_cmd = [
+            "gcloud", "auth", "activate-service-account", "--key-file",
+            acct_key_path]
+        self._subprocess_debug_wrap(gcloud_auth_cmd)
+
+    def gsutil_encrypt_cp_cmd(self, src: str, dst: str) -> None:
+        """ Copy a local file to a bucket with encryption """
+        gsutil_base_cmd = [self.gsutil_path,
+                           "-o", '"GSUtil:encryption_key={}"'.format(
+                               self.encrypt_key),
+                           "cp",  src, dst]
+
+        self._subprocess_debug_wrap(gsutil_base_cmd, True)
+        logger.info('Uploaded to encrypted bucket destination {}'.format(dst))
+
+    def rsync_cmd(self, src: str, dst: str, dry: bool = False) -> None:
+        """ Call gsutil rsync against two paths """
+        gsutil_base_cmd = [self.gsutil_path,
+                           "-m", "rsync", "-r", "-d", src, dst]
+
+        if dry:
+            gsutil_base_cmd.insert(4, "-n")
+
+        self._subprocess_debug_wrap(gsutil_base_cmd)
+
+    def rsync_source_bucket(self) -> str:
+        """ Pull down a copy of a bucket contents for local comparison"""
+
+        # Start out creating a temporary directory
+        self.tmp_dir = tempfile.mkdtemp()
+        logger.debug(
+            "Created local dir {}"
+            " for bucket manipulation".format(self.tmp_dir))
+
+        self.rsync_cmd(self.src, self.tmp_dir)
+
+        return self.tmp_dir
+
+    def tar_directory(self, tar_dir: str) -> str:
+        """ Tar+gzip up a directory and return path to that tar ball """
+        tar_file_path = os.path.join(tar_dir, "src-bucket.tar.gz")
+        tar = tarfile.open(tar_file_path, 'w:gz')
+        tar.add(tar_dir, arcname='')
+        tar.close()
+        logger.debug("Created tar at {} of {}".format(tar_file_path, tar_dir))
+
+        return tar_file_path
+
+    def upload_encrypted_timestamp_file(self, upload_file: str) -> None:
+        """ Given a file path upload said file to our backup bucket.
+            File is timestamp'd and includes file prefix. """
+        timestamp_name = datetime.datetime.now().strftime(
+            '%b-%d-%y-%X-{}.tar.gz'.format(self.suffix))
```

Comment:
Would strongly prefer `'%F-%R-%s` for the timestamp formatting, same as we use in the db backups.